pandas/core/construction.py

```diff
@@ -351,7 +351,9 @@ def array(
     return result
 
 
-def extract_array(obj: AnyArrayLike, extract_numpy: bool = False) -> ArrayLike:
+def extract_array(
+    obj: AnyArrayLike, extract_numpy: bool = False
+) -> Union[AnyArrayLike, ExtensionArray, np.ndarray]:
```

Comment:
i dont think we want AnyArrayLike in the return type, since we should never return Series or Index

Reply:
we will probably need to use an overload here eventually but not so easy when np.ndarray resolves to `Any` as we normally get messages about overlapping overloads.
```suggestion
) -> Union[Any, ExtensionArray, np.ndarray]:
```
for now.

Reply:
Thanks, looking at this more closely, the call to `array` always returns an EA. The possible types for input are EA, Index, Series, or np.ndarray, so that means the return can only be EA or np.ndarray. Let me know if this wrong, for now I've simply removed the `AnyArrayLike`.